modules-sdks/pax-terminal/instore-solution/NET/Methods/paymentasync.md

```diff
@@ -0,0 +1,196 @@
+%}
+
+### Parameters
+
+* **totalamount** - includes possible cashback amount
+* **cashback** - part of total amount that will be handed to customer
+* **currency** - currency code as a string representing ISO-4217 3 letter code. Has to be available in the terminal setup. The default is "SEK".
+
+### Returns
+
+A **PaymentRequestResult**
+
+A `PayementRequestResult.ResponseResult` of value `Success` means transaction approved.
+If `ResponseResult` is `Failure` there is an `ErrorCondition`. If `ErrorCondition` is `Busy`, wait awhile and try again.
+
+Make sure to always print the customer's receipt when available. For an aborted PaymentAsync there might not be one available.
+
+```c#
+public class NexoRequestResult
+{
+  public virtual string ResponseContent { get; set; }
+  public NexoResponseResult ResponseResult { get; set; }
+  public string ErrorCondition { get; set; }
+  public string ResponseText { get; set; }
+}
+
+public class PaymentRequestResult : NexoRequestResult
+{
+  public JObject CustomerReceiptData { get; set; }
+  public JObject MerchantReceiptData { get; set; }
+  public string FormattedReceipt { get; set; }
+  public string ReceiptBlob { get; set; }
+  public JObject SettlementData { get; set; }
+  public XElement OriginalTransaction { get; set; }
+  public string UICulture { get; set; }
+  public override string ResponseContent { get; set; }
+}
+
+
+```
+
+### CustomerReceiptData - Json object
+
+```json
+{
+ "Cardholder": {
+  "Mandatory": {
+   "Acquirer": {
+    "CardAcceptorNumber": "10020001",
+    "TerminalID": "877888"
+   },
+   "CardAcceptor": {
+    "Address1": "Kungsgatan 36",
+    "BankAgentName": "banky-bank",
+    "Name": "Demo shop",
+    "OrganisationNumber": "5565671-6165",
+    "PostZipCode": "5050",
+    "TownCity": "merchant-Base24-City"
+   },
+   "CardDetails": {
+    "ApplicationIdentifier": "A0000000031010",
+    "CardSchemeName": {
+     "ApplicationLabel": "Visa Debit"
+    },
+    "PrimaryAccountNumber": "************4565",
+    "TerminalVerificationResult": "0080008000",
+    "TransactionStatusInformation": "E800"
+   },
+   "Outcome": {
+    "ApprovalCode": "942932",
+    "AuthorisationResponder": "9",
+    "AuthorisationResponseCode": "00",
+    "DebitStatus": "00"
+   },
+   "Payment": {
+    "AuthorisationChannel": "1",
+    "CardholderVerificationMethod": "a",
+    "Currency": "SEK",
+    "FinancialInstitution": "SWE",
+    "PaymentAmount": "41,00",
+    "ReceiptNumber": "8778880241",
+    "SignatureBlock": false,
+    "TotalAmount": "41,00",
+    "TransactionSource": "C",
+    "TransactionType": "00"
+   },
+   "TimeStamp": {
+    "DateOfPayment": "2023-03-16",
+    "TimeOfPayment": "18:01"
+   }
+  },
+  "Optional": {
+   "CardAcceptor": {
+    "CountryName": "752",
+    "PhoneNumber": "+4684051000"
+   },
+   "CardDetails": {
+    "CardSchemeName": {
+     "ApplicationLabel": "Visa Debit"
+    }
+   },
+   "Payment": {
+    "Reference": "1801042"
+   }
+  }
+ }
+}
+```
```

Comment:
Could be nice to show additional APM receipt details but can do that in a future release

Reply:
absolutely, but since it is not fully available yet I didn't mention APM in the examples.